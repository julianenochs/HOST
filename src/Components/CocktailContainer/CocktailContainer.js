import React from 'react';
import './CocktailContainer.scss';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { selectCocktail } from '../../Actions/index';

const CocktailContainer = (props) => {
  console.log(props);
  return (<div className='cocktail__container'>
  {props.cocktailsByName.map((cocktail, i) => {
    return <div className='cocktail__element'>
        <ul className='cocktail-name' key={i}>{cocktail.strDrink}</ul>
        <img className='cocktail__img'src={`${cocktail.strDrinkThumb}`} alt={`${cocktail.strDrink}`}/>
        <Link to={`/drinks/${cocktail.idDrink}`}>
          <button className='view-cocktail__button' onClick={() => props.selectCocktail(cocktail)}> View Cocktail </button>
        </Link>
      </div>
  })}
    </div>)
}

const mapStateToProps = state => ({
  cocktailsByName: state.cocktailsByName,
});

const mapDispatchToProps = dispatch => ({
  selectCocktail: (cocktail) => dispatch(selectCocktail(cocktail))
})

export default connect(mapStateToProps, mapDispatchToProps)(CocktailContainer);