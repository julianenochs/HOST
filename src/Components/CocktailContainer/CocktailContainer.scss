.cocktail__container {
  display: flex;
  flex-wrap: wrap;
  font-family: 'Federo', sans-serif;
  font-size: 1.3em;
  height: overflow;
  overflow: scroll;
  opacity: 0.8;
  width: 85%;
}

.cocktail-name {
  font-size: 1.3em;
  height: 5vh;
  margin-left: -40px;
}

.cocktail__element {
  align-items: center;
  background-color: rgba(92, 92, 92, .85);
  border-radius: 5px;
  color: white;
  display: flex;
  flex-direction: column;
  height: 45%;
  margin: 1%;
  text-align: center;
  width: 14%;
    &:hover {
    background-color: rgba(92, 92, 92, 1);
  }
}

.cocktail__img {
  height: 150px;
  width: 150px;
}

.view-cocktail__button {
  background-color:  rgba(92, 92, 92, 0);
  border: none;
  color: white;
  font-family: 'Federo', sans-serif;
  font-size: 1em;
  margin-top: 15%;
}