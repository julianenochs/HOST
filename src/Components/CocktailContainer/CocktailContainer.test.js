import React from 'react';
import { shallow } from 'enzyme';
import { CocktailContainer } from './CocktailContainer';

describe('Cocktail Container', () => {
  let wrapper;
  let mockSetCurrentCocktail = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<CocktailContainer 
      setCurrentCocktail={mockSetCurrentCocktail} />)
  });

  it.skip('should match the Cocktail Container snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})