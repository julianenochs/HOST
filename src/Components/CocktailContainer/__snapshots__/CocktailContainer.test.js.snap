// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cocktail Container should match the Cocktail Container snapshot 1`] = `undefined`;
